scripts/zones/Chateau_dOraguille/npcs/Aramaviont.lua

```diff
@@ -3,12 +3,53 @@
 --  NPC: Aramaviont
 -- Standard Info NPC
 -----------------------------------
+local ID = require("scripts/zones/Chateau_dOraguille/IDs")
+-----------------------------------
 
 function onTrade(player, npc, trade)
 end
 
 function onTrigger(player, npc)
-    player:startEvent(518)
+
+    -- This NPC is relevant only to San d'Orians on missions
+    if player:getNation() ~= tpz.nation.SANDORIA then
+        player:startEvent(518)
+
+    else
+        local missions = tpz.mission.id.sandoria
+        local currentMission = player:getCurrentMission(SANDORIA)
+        local missionStatus = player:getCharVar("MissionStatus")
+
+        -- San d'Oria Rank 10 (optional dialogue)
+        if player:getCharVar("SandoEpilogue") == 1 then
+            player:startEvent(11)
+
+        -- San d'Oria 9-2 "The Heir to the Light" (optional dialogue)
+        elseif currentMission == missions.THE_HEIR_TO_THE_LIGHT and missionStatus > 1 then
+            if missionStatus > 4 then
+                player:startEvent(14)
+            else
+                player:startEvent(13)
+            end
+
+        -- San d'Oria 8-2 "Lightbringer" (optional dialogue)
+        elseif player:hasCompletedMission(SANDORIA, missions.LIGHTBRINGER) and
+            player:getRank() == 9 and player:getRankPoints() == 0
+        then
```

Comment:
Style request:
```lua
        elseif
            player:hasCompletedMission(SANDORIA, missions.LIGHTBRINGER) and
            player:getRank() == 9 and player:getRankPoints() == 0
        then
```
Since its 2 wide to be one line, this is the preferred way to break them in lua

Reply:
Ah that's even better. Will do tomorrow.